core/Translation/Transifex/API.php

```diff
@@ -0,0 +1,143 @@
+<?php
+/**
+ * Piwik - free/libre analytics platform
+ *
+ * @link http://piwik.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ *
+ */
+namespace Piwik\Translation\Transifex;
+
+use Exception;
+use Piwik\Exception\AuthenticationFailedException;
+
+class API
+{
+    protected $apiUrl = 'https://www.transifex.com/api/2/';
+    protected $username = '';
+    protected $password = '';
+    protected $projectSlug = '';
+
+    public function __construct($username, $password, $project='piwik')
+    {
+        $this->username = $username;
+        $this->password = $password;
+        $this->projectSlug = $project;
+    }
+
+    /**
+     * Returns all resources available on Transifex project
+     *
+     * @return array
+     */
+    public function getAvailableResources()
+    {
+        static $resources;
```

Comment:
Those should be class properties, static means they will be shared across instances